woocommerce/woocommerce-admin
# Inbox notification: Added layout changes

Fixes #partially-4099

This PR adds:
- 2 new columns to the table `wp_wc_admin_notes `: `layout` and `image`.
- Also, these two props were added to the Note class and in the Notes' API calls.
- Unit tests were modified to adapt them to the changes.

_There are some buttons that look weird because there are some inherited styles that modify them. As soon as they change the buttons will look normal again._

> **_NOTE:_**  As this is a part of a big functionality this branch will be merged with the branch `add/4099`, and it will be that branch that will be merged with `master`.
To make the code reviewing easier, this PR only has the code referring to the functionality described in the title.
If you want to test everything together, all the changes are merged in the branch: `add/4099_test_branch`.

### Screenshots
It should look like:
![image](https://user-images.githubusercontent.com/22080/79793339-41309700-8305-11ea-877b-05fe4a912293.png)

And the flow is something like this:
![Screen Capture on 2020-05-19 at 11-54-33](https://user-images.githubusercontent.com/1314156/82342221-bf955d00-99c7-11ea-82e7-6eee6e495d04.gif)

### Detailed test instructions:
- dbDelta should add the `layout` and `image` columns to the table `wp_wc_admin_notes`
- After having the database updated you should add an image to the notes. Below I will leave an SQL to add 2 notes with the 2 types of images we need:
````
INSERT INTO `wp_wc_admin_notes` (`name`, `type`, `locale`, `title`, `content`, `icon`, `content_data`, `status`, `source`, `date_created`, `date_reminder`, `is_snoozable`, `layout`, `image`) VALUES
('wc-admin-task-list-complete', 'info', 'en_US', 'Congratulations - your store is ready for launch!', 'You’re ready to take your first order - may the sales roll in!', 'notice', '{}', 'unactioned', 'woocommerce-admin', '2020-03-10 18:38:10', NULL, 0, 'thumbnail', 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTlm2mX1h6GfMnT9sDD9GsJB7of3MZw3UDzCAQM9EBepiJzyxXH&usqp=CAU'),
('wc-admin-marketing-intro', 'info', 'en_US', 'Connect with your audience', 'Grow your customer base and increase your sales with marketing tools built for WooCommerce.', 'speaker', '{}', 'unactioned', 'woocommerce-admin', '2020-04-09 16:08:41', NULL, 0, 'banner', 'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSAK9Xg4L6FGmDAW5UVtVEv1IXKtGV3-rxYLfAzOBF-fMUdmyWz&usqp=CAU');
````
- To be able to see the inbox notifications, press `Inbox` in the navigation bar.
- The list of inbox notifications should be visible.
- There are 3 types of messages: `banner`, `thumbnail` and `plain`. Verify that every type looks as expected. 
- Verify that the messages have the right style for `read` and `unread` messages. To do this it's necessary to add a new message or change the `woocommerce_admin_activity_panel_inbox_last_read` in the `wp_usermeta` table.
You can use a sentence like this:
````
UPDATE `wp_usermeta` SET `meta_value` = '1581364679000' WHERE meta_key = 'woocommerce_admin_activity_panel_inbox_last_read'
````

- Verify that a dropdown with the options: `Dismiss this message` and `Dismiss all messages` is visible after pressing `Dissmis` for all the messages.
- Verify that a confirmation modal is visible after pressing any of the two dropdown options.



### Changelog Note:

Dev: Inbox notification: Added layout changes
